CMakeLists.txt

```diff
@@ -23,11 +23,22 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/bin")
 
 if(NOT CMAKE_BUILD_TYPE)
   message(STATUS "No CMAKE_BUILD_TYPE given, default to Release")
-  set(CMAKE_BUILD_TYPE Release CACHE STRING
+  set(CMAKE_BUILD_TYPE Debug CACHE STRING
     "Set the build type. Available values are: Debug Release RelWithDebInfo MinSizeRel"
     FORCE)
 endif()
 
+if(CMAKE_BUILD_TYPE STREQUAL Debug)
+  message(STATUS "Build in Debug mode")
+  # Add -O0 to remove optimizations for debugging.
+  set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -O0")
```

Comment:
thanks... @csukuangfj does this look OK?

Reply:
It looks fine to me.

---

I would suggest saving `CMAKE_CXX_FLAGS_DEBUG` to cmake cache, i.e.,

```cmake
set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -O0" CACHE STRING "" FORCE)
```

To make it more robust, we can convert `CMAKE_BUILD_TYPE` to lowercase or uppercase and compare it against `debug` or `DEBUG`.